staker/block_validator.go

```diff
@@ -327,6 +351,27 @@ func NewBlockValidator(
 			PosInBatch: 0,
 		}
 	}
+	if config().Dangerous.Revalidation.StartBlock > 0 {
+		startBlock := config().Dangerous.Revalidation.StartBlock
+		messageCount, err := inbox.GetBatchMessageCount(startBlock - 1)
+		if err != nil {
+			return nil, err
+		}
+		res, err := streamer.ResultAtMessageIndex(messageCount)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
This should actually be something like:
```suggestion
                if messageCount == 0 {
                        return nil, fmt.Errorf("messageCount is zero")
                }
                msgIdx := messageCount - 1
		res, err := streamer.ResultAtMessageIndex(msgIdx)
		if err != nil {
			return nil, err
		}
```

Reply:
Done.